src/url.cpp

```diff
@@ -122,16 +122,37 @@ namespace Url
             if (index != std::string::npos)
             {
                 query_ = path_.substr(index + 1);
+                remove_repeats(query_, '?');
+                remove_repeats(query_, '&');
                 path_.resize(index);
             }
 
             index = path_.find(';');
             if (index != std::string::npos)
             {
                 params_ = path_.substr(index + 1);
+                remove_repeats(params_, ';');
                 path_.resize(index);
             }
         }
     }
 
+    void Url::remove_repeats(std::string& str, const char chr)
+    {
+        size_t dest = 0;
+        bool seen = true;
+        for (size_t src = 0; src < str.length(); ++src)
+        {
+            if (!seen || (str[src] != chr))
+            {
+                str[dest] = str[src];
```

Comment:
I understand it is not *always* true, but I find it odd to do the assignment when it *is* true. I almost prefer adding the check just for clarity.